### PR TITLE
Improve test coverage Bio.UniProt

### PR DESCRIPTION
**Pull request by svalqui**
This pull request addresses issue #1288 

- [X] I hereby agree to dual licence this and any previous contributions under both
the _Biopython License Agreement_ **AND** the _BSD 3-Clause License_.

- [X] I have read the ``CONTRIBUTING.rst`` file, have run ``flake8`` locally, and
understand that AppVeyor and TravisCI will be used to confirm the Biopython unit
tests and style checks pass with these changes.

- [X] I have added my name to the alphabetical contributors listings in the files
``NEWS.rst`` and ``CONTRIB.rst`` as part of this pull request, am listed
already, or do not wish to be listed. (*This acknowledgement is optional.*)

Regarding GOA.py
I have removed 2 items('Annotation_Target_Set', 'GO_Annotation_Complete') from GPI11FIELDS, as they are not included in the GPI 1.1 Format.

I have added samples files from ftp://ftp.ebi.ac.uk/pub/databases/GO/goa, they have been trimmed to 300 records each as otherwise they are bigger than 1Mb.

### CODE REVIEW

File: Bio/UniProt/GOA.py [hunk 1]
```diff
@@ -118,9 +120,7 @@
     'Taxon',
     'Parent_Object_ID',
     'DB_Xref',
-    'Gene_Product_Properties',
-    'Annotation_Target_Set',
-    'GO_Annotation_Complete']
+    'Gene_Product_Properties']
```
Comment (peterjc) [inline Bio/UniProt/GOA.py]: Was this an oversight in the original, and the removed fields were GPI 1.0 only?

**Comment by peterjc** [comment]
@idoerg do you have time to look this over as it is based on one of your more recent contributions to Biopython?

**Comment by peterjc** [comment]
The tests are failing on AppVeyor (Windows):

```
======================================================================
FAIL: test_gaf_iterator (test_UniProt_GOA.GoaTests)
Test GOA GAF file iterator.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\biopython\Tests\test_UniProt_GOA.py", line 31, in test_gaf_iterator
    self.assertEqual(list(recs[0].keys()), GOA.GAF20FIELDS)
AssertionError: Lists differ: ['Synonym', 'Qualifier', 'Date... != ['DB', 'DB_Object_ID', 'DB_Obj...
First differing element 0:
'Synonym'
'DB'
- ['Synonym',
+ ['DB',
+  'DB_Object_ID',
+  'DB_Object_Symbol',
   'Qualifier',
+  'GO_ID',
+  'DB:Reference',
+  'Evidence',
+  'With',
+  'Aspect',
+  'DB_Object_Name',
+  'Synonym',
+  'DB_Object_Type',
+  'Taxon_ID',
   'Date',
   'Assigned_By',
-  'DB_Object_Name',
   'Annotation_Extension',
-  'DB',
-  'GO_ID',
-  'Taxon_ID',
-  'DB:Reference',
-  'DB_Object_Type',
-  'Aspect',
-  'DB_Object_ID',
-  'DB_Object_Symbol',
-  'With',
-  'Gene_Product_Form_ID',
?                        ^
+  'Gene_Product_Form_ID']
?                        ^
-  'Evidence']
======================================================================
FAIL: test_gpa_iterator (test_UniProt_GOA.GoaTests)
Test GOA GPA file iterator.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\biopython\Tests\test_UniProt_GOA.py", line 69, in test_gpa_iterator
    self.assertEqual(list(recs[0].keys()), GOA.GPA11FIELDS)
AssertionError: Lists differ: ['ECO_Evidence_code', 'Qualifi... != ['DB', 'DB_Object_ID', 'Qualif...
First differing element 0:
'ECO_Evidence_code'
'DB'
- ['ECO_Evidence_code',
+ ['DB',
+  'DB_Object_ID',
   'Qualifier',
+  'GO_ID',
+  'DB:Reference',
+  'ECO_Evidence_code',
+  'With',
+  'Interacting_taxon_ID',
+  'Date',
   'Assigned_by',
   'Annotation Extension',
-  'Annotation_Properties',
?                         ^
+  'Annotation_Properties']
?                         ^
-  'DB',
-  'GO_ID',
-  'Date',
-  'DB:Reference',
-  'DB_Object_ID',
-  'Interacting_taxon_ID',
-  'With']
======================================================================
FAIL: test_gpi_iterator (test_UniProt_GOA.GoaTests)
Test GOA GPI file iterator.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "C:\projects\biopython\Tests\test_UniProt_GOA.py", line 91, in test_gpi_iterator
    self.assertEqual(list(recs[0].keys()), GOA.GPI11FIELDS)
AssertionError: Lists differ: ['Gene_Product_Properties', 'D... != ['DB_Object_ID', 'DB_Object_Sy...
First differing element 0:
'Gene_Product_Properties'
'DB_Object_ID'
- ['Gene_Product_Properties',
+ ['DB_Object_ID',
+  'DB_Object_Symbol',
   'DB_Object_Name',
+  'DB_Object_Synonym',
+  'DB_Object_Type',
   'Taxon',
   'PareSkipping any tests requiring internet access
Python version: 2.7.16 |Anaconda, Inc.| (default, Mar 14 2019, 15:49:40) [MSC v.1500 32 bit (Intel)]
Operating system: nt win32
nt_Object_ID',
-  'DB_Object_Symbol',
   'DB_Xref',
+  'Gene_Product_Properties']
-  'DB_Object_Synonym',
-  'DB_Object_Type',
-  'DB_Object_ID']
----------------------------------------------------------------------
```

My guess is you need to do a sort somewhere, perhaps you are seeing dictionary order differences here?

File: Bio/UniProt/GOA.py (hunk 1 above)
Comment (svalqui) [reply Bio/UniProt/GOA.py]: I think so, the documentation  for GPI 1.0 and GPI 1.1 is no longer available only for GPI 1.2
http://geneontology.org/docs/gene-product-information-gpi-format/
the file, gp_information.goa_yeast.28.gpi, added here dated 2013-11-11, includes the fields on its header, that is what I have used as a reference.

**Comment by peterjc** [comment]
About to rebase and fix merge conflict (quote style)...

**Pull request merged**